cmd/vcfFilter/vcfFilter.go

```diff
@@ -4,25 +4,61 @@ import (
 	"flag"
 	"fmt"
 	"github.com/vertgenlab/gonomics/fileio"
+	"github.com/vertgenlab/gonomics/popgen"
 	"github.com/vertgenlab/gonomics/vcf"
 	"log"
 	"math"
 	"strings"
 )
 
-func vcfFilter(infile string, outfile string, chrom string, minPos int, maxPos int, ref string, alt string, minQual float64) {
-	ch, _ := vcf.GoReadToChan(infile)
+func vcfFilter(infile string, outfile string, groupFile string, chrom string, minPos int, maxPos int, ref string, alt string, minQual float64) {
+	ch, header := vcf.GoReadToChan(infile)
 	out := fileio.EasyCreate(outfile)
 	defer out.Close()
 	altSlice := strings.Split(alt, ",")
 
+	var samplesToKeep []int = make([]int, 0) //this var holds all of the indices from samples (defined below as the sample list in the header) that we want to keep in the output file.
+
+	if groupFile != "" {
+		groups := popgen.ReadGroups(groupFile)
+		samples := vcf.HeaderGetSampleList(header)
+
+		for i := 0; i < len(samples); i++ {
+			if popgen.GroupsContains(groups, samples[i]) {
+				samplesToKeep = append(samplesToKeep, i)
+			}
+		}
+		outSamples := filterHeaderSamplesToKeep(samples, samplesToKeep)
+		vcf.HeaderUpdateSampleList(header, outSamples)
+	}
+
+	vcf.NewWriteHeader(out.File, header)
+
 	for v := range ch {
 		if vcf.Filter(v, chrom, minPos, maxPos, ref, altSlice, minQual) {
+			v.Samples = filterRecordsSamplesToKeep(v.Samples, samplesToKeep)
 			vcf.WriteVcf(out.File, v)
 		}
 	}
 }
 
+func filterRecordsSamplesToKeep(recordSamples []vcf.GenomeSample, samplesToKeep []int) []vcf.GenomeSample {
+	var answer []vcf.GenomeSample
+	for i := 0; i < len(samplesToKeep); i++ {
```

Comment:
Could use the "range" syntax here.  I think it would be:
for i := range samplesToKeep {